File: java/804_UniqueMorseCodeWords.java
class Solution {
    String[] codes = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
    public int uniqueMorseRepresentations(String[] words) {
        HashSet<String> uniques = new HashSet<String>();
        for (String word : words) {
            String currCode = "";
            for (Character c : word.toCharArray()) {
                currCode += codes[c - 'a'];
            }
            uniques.add(currCode);
        }
        return uniques.size();
    }
}

File: python/804_UniqueMorseCodeWords.py
class Solution:
    codes = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        return len({ ''.join([self.codes[ord(c) - ord('a')] for c in word]) for word in words })
